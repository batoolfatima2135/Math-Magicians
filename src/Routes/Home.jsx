const Home = () => (
  <div className="px-2">
    <h2 className="py-4 text-success">Welcome to our Page!</h2>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.Officia quaerat soluta aliquam
      veritatis recusandae, reiciendis quo ullam cum non molestias voluptates saepe at nisi
      earum, et reprehenderit porro magnam. Lorem ipsum dolor sit amet consectetur adipisicing
      elit. Officiis facere consectetur minus. Laborum ex blanditiis illo nobis, reprehenderit
      porro! Nemo ipsum vero quos sed! Ea quidem suscipit quisquam blanditiis distinctio.
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque
      soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam
      alias quasi! Praesentium, qui explicabo.
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque
      soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam
      alias quasi! Praesentium, qui explicabo.
    </p>
    <p>
      Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus placeat, veritatis omnis,
      laborum magnam quia, ex nisi est vero saepe porro? Lorem ipsum dolor sit amet consectetur
      adipisicing elit. Expedita, aut incidunt molestiae hic natus officiis alias sint placeat aperi
      ea dolor et exercitationem quo tempore quas eligendi qui vero deserunt!
      Blanditiis, accusantium. Mollitia alias, dolor reiciendis accusantium saepe amet!
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque
      soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam
      alias quasi! Praesentium, qui explicabo.
      Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque
      soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam
      alias quasi! Praesentium, qui explicabo.
    </p>
  </div>
);
export default Home;
