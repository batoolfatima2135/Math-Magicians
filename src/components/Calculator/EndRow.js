// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';
import calculate from '../logic/calculate';

function EndRow(props) {
  const { items, state, updateState } = props;
  const getdata = (event) => {
    const btn = event.target.textContent;
    updateState(calculate(state, btn));
  };
  return (
    <div className="row">
      <button type="button" key={items[0]} onClick={getdata} className="col-6 button text-center py-3 border">{ items[0] }</button>
      <button type="button" key={items[1]} onClick={getdata} className="col-3 button  text-center py-3 border">{ items[1] }</button>
      <button type="button" key={items[2]} onClick={getdata} className="col-3 text-center py-3 border last-item">{ items[2] }</button>

    </div>
  );
}
EndRow.propTypes = {
  items: PropTypes.arrayOf(PropTypes.string).isRequired,
  state: PropTypes.shape({
    total: PropTypes.string,
    next: PropTypes.string,
    operation: PropTypes.string,
  }).isRequired,
  updateState: PropTypes.func.isRequired,
};

export default EndRow;
