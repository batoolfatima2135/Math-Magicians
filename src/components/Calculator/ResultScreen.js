// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

function ResultScreen(props) {
  const { answere } = props;
  return (
    <div className="row">
      <div className="col-12 bg-secondary text-light py-3 text-end">{answere}</div>
    </div>
  );
}
ResultScreen.propTypes = { answere: PropTypes.number };
ResultScreen.defaultProps = { answere: 0 };
export default ResultScreen;
