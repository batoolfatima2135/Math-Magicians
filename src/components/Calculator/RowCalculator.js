// eslint-disable-next-line import/no-extraneous-dependencies
import PropTypes from 'prop-types';

function RowCalculator(props) {
  const { items } = props;
  return (
    <div className="row">
      { items.map((item, index) => (
        <div key={item} className={`col-3 button text-center py-3 border ${index % 4 === 3 ? 'last-item' : ''}`}>{item}</div>
      ))}
    </div>
  );
}
RowCalculator.propTypes = { items: PropTypes.arrayOf(PropTypes.string).isRequired };

export default RowCalculator;
