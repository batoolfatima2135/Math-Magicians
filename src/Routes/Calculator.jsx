import React, { useState } from 'react';
import RowCalculator from '../components/Calculator/RowCalculator';
import ResultScreen from '../components/Calculator/ResultScreen';
import EndRow from '../components/Calculator/EndRow';

const Calculator = () => {
  const [calculatorData, setCalculatorData] = useState({
    total: null,
    next: null,
    operation: null,
  });
  const updateState = (newState) => {
    setCalculatorData(newState);
  };
  return (
    <div className="row justify-content-center">
      <div className="col-lg-4 col-md-6 col-6 mt-5">
        <h3 className="my-2 text-success">Lets do some maths!</h3>
      </div>
      <div className="col-lg-6 col-md-6 col-6 mt-5 border border-5 border-success">
        <ResultScreen
          next={calculatorData.next}
          operation={calculatorData.operation}
          total={calculatorData.total}
        />
        <RowCalculator items={['AC', '+/-', '%', '\u00F7', '7', '8', '9', 'x', '4', '5', '6', '-', '1', '2', '3', '+']} state={calculatorData} updateState={updateState} />
        <EndRow items={['0', '.', '=']} state={calculatorData} updateState={updateState} />
      </div>
    </div>
  );
};

export default Calculator;
