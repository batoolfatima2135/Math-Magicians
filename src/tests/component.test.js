import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Quotes from '../Routes/Quote';
import Home from '../Routes/Home';
import Calculator from '../Routes/Calculator';

describe('Jest snapshots to test the components.', () => {
  test('Quote component renders correctly', () => {
    const tree = renderer.create(<Quotes />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Home component renders correctly', () => {
    const tree = renderer.create(<Home />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Calculator component renders correctly', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ResultScreen component renders correctly', () => {
    const tree = renderer.create(<ResultScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('EndRow component renders correctly', () => {
    const state = { total: '5', next: null, operation: null };
    const updateState = jest.fn();
    const items = ['1', '2', '3'];
    const tree = renderer.create(<EndRow
      items={items}
      state={state}
      updateState={updateState}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('RowCalculator component renders correctly', () => {
    const state = { total: '5', next: null, operation: null };
    const updateState = jest.fn();
    const items = ['1', '2', '3'];
    const tree = renderer.create(<RowCalculator
      items={items}
      state={state}
      updateState={updateState}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
