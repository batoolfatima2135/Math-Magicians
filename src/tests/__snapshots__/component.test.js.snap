// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest snapshots to test the components. Calculator component renders correctly 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-lg-4 col-md-6 col-6 mt-5"
  >
    <h3
      className="my-2 text-success"
    >
      Lets do some maths!
    </h3>
  </div>
  <div
    className="col-lg-6 col-md-6 col-6 mt-5 border border-5 border-success"
  >
    <div
      className="row"
    >
      <div
        className="col-12 bg-success text-light py-3 text-end fw-bold"
      >
        0
      </div>
    </div>
    <div
      className="row"
    >
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border last-item"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border last-item"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border last-item"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border "
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="col-3 button text-center fw-bold py-3 border last-item"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="row"
    >
      <button
        className="col-6 button fw-bold text-center py-3 border"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="col-3 button fw-bold  text-center py-3 border"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="col-3 text-center fw-bold  py-3 border last-item"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Jest snapshots to test the components. EndRow component renders correctly 1`] = `
<div
  className="row"
>
  <button
    className="col-6 button fw-bold text-center py-3 border"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="col-3 button fw-bold  text-center py-3 border"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="col-3 text-center fw-bold  py-3 border last-item"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
</div>
`;

exports[`Jest snapshots to test the components. Home component renders correctly 1`] = `
<div
  className="px-2"
>
  <h2
    className="py-4 text-success"
  >
    Welcome to our Page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit.Officia quaerat soluta aliquam veritatis recusandae, reiciendis quo ullam cum non molestias voluptates saepe at nisi earum, et reprehenderit porro magnam. Lorem ipsum dolor sit amet consectetur adipisicing elit. Officiis facere consectetur minus. Laborum ex blanditiis illo nobis, reprehenderit porro! Nemo ipsum vero quos sed! Ea quidem suscipit quisquam blanditiis distinctio. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam alias quasi! Praesentium, qui explicabo. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam alias quasi! Praesentium, qui explicabo.
  </p>
  <p>
    Lorem ipsum, dolor sit amet consectetur adipisicing elit. Temporibus placeat, veritatis omnis, laborum magnam quia, ex nisi est vero saepe porro? Lorem ipsum dolor sit amet consectetur adipisicing elit. Expedita, aut incidunt molestiae hic natus officiis alias sint placeat aperi ea dolor et exercitationem quo tempore quas eligendi qui vero deserunt! Blanditiis, accusantium. Mollitia alias, dolor reiciendis accusantium saepe amet! Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam alias quasi! Praesentium, qui explicabo. Lorem ipsum dolor sit amet, consectetur adipisicing elit. Velit a blanditiis ipsa cumque soluta vitae doloribus ipsum! Laudantium hic voluptas iusto aliquam odio, doloremque ipsam alias quasi! Praesentium, qui explicabo.
  </p>
</div>
`;

exports[`Jest snapshots to test the components. Quote component renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row justify-content-center align-items-center"
    style={
      Object {
        "height": "500px",
      }
    }
  >
    <div
      className="col-12 align-self-center text-center"
    >
      <q
        className="h3"
      >
        <b>
          new
        </b>
      </q>
    </div>
  </div>
</div>
`;

exports[`Jest snapshots to test the components. ResultScreen component renders correctly 1`] = `
<div
  className="row"
>
  <div
    className="col-12 bg-success text-light py-3 text-end fw-bold"
  >
    0
  </div>
</div>
`;

exports[`Jest snapshots to test the components. RowCalculator component renders correctly 1`] = `
<div
  className="row"
>
  <button
    className="col-3 button text-center fw-bold py-3 border "
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="col-3 button text-center fw-bold py-3 border "
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="col-3 button text-center fw-bold py-3 border "
    onClick={[Function]}
    type="button"
  >
    3
  </button>
</div>
`;
